src/txHashBuilder.c

```diff
@@ -387,6 +447,38 @@ void txHashBuilder_addOutput_datumHash(
 	builder->state = TX_HASH_BUILDER_IN_OUTPUTS;
 }
 
+void txHashBuilder_addOutput_datumOption(
+        tx_hash_builder_t* builder,
+        const uint8_t* bytesBuffer, size_t bufferSize, uint8_t datumOption, bool includeScriptref
```

Comment:
We want an enum for better readability for `datumOption`, and I'd pass it before the buffer. Also, rename bytesBuffer -> buffer for consistency.

Reply:
Passing `includeScriptref` as an argument for the sole purpose of changing the internal txhashbuilder state smells like a bad design: it is not related to datumOption.

Is this approach copied from somewhere? Perhaps we should rethink the internal builder state. I'd expect only `cbor_append_post_alonzo_txOutput` to receive `includeScriptref`. Maybe we don't have a place to store the information in `tx_hash_builder_t`?